src/onyx/messaging/aeron.clj

```diff
@@ -17,6 +17,38 @@
              [java.util.function Consumer]
              [java.util.concurrent TimeUnit]))
 
+(def max-port 50000)
+
+(defrecord AeronPeerGroup [opts]
+  component/Lifecycle
+  (start [component]
+    (taoensso.timbre/info "Starting Aeron")
+    (let [media-driver (MediaDriver/launch)]
+      (assoc component :media-driver media-driver)))
+
+  (stop [{:keys [media-driver] :as component}]
+    (.close media-driver)
+    (assoc component :media-driver nil)))
+
+(defn aeron-peer-group [opts]
+  (map->AeronPeerGroup {:opts opts}))
+
+(defmethod extensions/assign-site-resources :aeron
+  [config peer-site peer-sites]
+  (let [start-port (:aeron/start-port config)
+        existing-ports (->> (vals peer-sites)
+                            (filter 
+                              (fn [s]
+                                (and (= (:aeron/bind-addr peer-site) 
+                                        (:aeron/bind-addr s))
+                                     (= (:aeron/external-addr peer-site) 
+                                        (:aeron/external-addr s)))))
+                            (map :aeron/port)
+                            set)
```

Comment:
@MichaelDrogalis I *think* allocating by unique bind-addr/external-addr combinations makes sense. If you have a different external-addr to your bind-addr it's probably safe to use the same port because they might be just two different containers which use the same local address.